lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -1406,11 +1407,16 @@ void DAP::EventThread() {
             // avoids sending paths that should be source mapped. Note that
             // CreateBreakpoint doesn't apply source mapping and certain
             // implementation ignore the source part of this event anyway.
-            llvm::json::Value source_bp = bp.ToProtocolBreakpoint();
-            source_bp.getAsObject()->erase("source");
+            protocol::Breakpoint protocol_bp = bp.ToProtocolBreakpoint();
+            llvm::json::Value protocol_bp_value = toJSON(protocol_bp);
+
+            // "source" is not needed here, unless we add adapter data to be
+            // saved by the client.
+            if (protocol_bp.source && !protocol_bp.source->adapterData)
+              protocol_bp_value.getAsObject()->erase("source");
```

Comment:
Instead of converting this to a Value, should we just set the `protocol_bp.source = std::nullopt;`?

Reply:
:100: